ts/session/utils/job_runners/jobs/GroupSyncJob.ts

```diff
@@ -0,0 +1,325 @@
+  extraStoreRequests,
+}: WithGroupPubkey &
+  WithRevokeSubRequest & {
+    supplementalKeysSubRequest?: StoreGroupKeysSubRequest;
+    deleteAllMessagesSubRequest?: DeleteAllFromGroupMsgNodeSubRequest;
+    extraStoreRequests: Array<StoreGroupMessageSubRequest>;
+  }): Promise<RunJobResult> {
+  // save the dumps to DB even before trying to push them, so at least we have an up to date dumps in the DB in case of crash, no network etc
+  await LibSessionUtil.saveDumpsToDb(groupPk);
+  const { allOldHashes, messages: pendingConfigData } =
+    await LibSessionUtil.pendingChangesForGroup(groupPk);
+  // If there are no pending changes nor any requests to be made,
+  // then the job can just complete (next time something is updated we want
+  // to try and run immediately so don't schedule another run in this case)
+  if (
+    isEmpty(pendingConfigData) &&
+    isEmpty(supplementalKeysSubRequest) &&
+    isEmpty(revokeSubRequest) &&
+    isEmpty(unrevokeSubRequest) &&
+    isEmpty(deleteAllMessagesSubRequest) &&
+    isEmpty(extraStoreRequests)
+  ) {
+    window.log.debug(`pushChangesToGroupSwarmIfNeeded: ${ed25519Str(groupPk)}: nothing to push`);
+    return RunJobResult.Success;
+  }
+
+  const group = await UserGroupsWrapperActions.getGroup(groupPk);
+  if (!group) {
+    window.log.debug(`pushChangesToGroupSwarmIfNeeded: ${ed25519Str(groupPk)}: group not found`);
+    return RunJobResult.Success;
+  }
+
+  if (window.sessionFeatureFlags.debug.debugLibsessionDumps) {
+    const dumps = await MetaGroupWrapperActions.metaMakeDump(groupPk);
+    window.log.info(
+      `pushChangesToGroupSwarmIfNeeded: current meta dump: ${ed25519Str(groupPk)}:`,
+      to_hex(dumps)
+    );
+  }
+
+  const pendingConfigRequests = StoreGroupRequestFactory.makeStoreGroupConfigSubRequest({
+    group,
+    pendingConfigData,
+  });
+
+  const deleteHashesSubRequest = DeleteGroupHashesFactory.makeGroupHashesToDeleteSubRequest({
+    group,
+    messagesHashes: allOldHashes,
+  });
+  const extraRequests = compact([
+    deleteHashesSubRequest,
+    revokeSubRequest,
+    unrevokeSubRequest,
+    deleteAllMessagesSubRequest,
+  ]);
+
+  const extraRequestWithExpectedResults = extraRequests.filter(
+    m =>
+      m instanceof SubaccountRevokeSubRequest ||
+      m instanceof SubaccountUnrevokeSubRequest ||
+      m instanceof DeleteAllFromGroupMsgNodeSubRequest ||
+      m instanceof DeleteHashesFromGroupNodeSubRequest
+  );
+
+  const sortedSubRequests = compact([
+    supplementalKeysSubRequest, // this needs to be stored first
+    ...pendingConfigRequests, // groupKeys are first in this array, so all good, then groupInfos are next
+    ...extraStoreRequests, // this can be stored anytime
+    ...extraRequests,
+  ]);
+
+  const result = await MessageSender.sendEncryptedDataToSnode({
+    // Note: this is on purpose that supplementalKeysSubRequest is before pendingConfigRequests
+    // as this is to avoid a race condition where a device is polling right
```

Comment:
Changed to be 

```ts
    // Note: this is on purpose that supplementalKeysSubRequest is before pendingConfigRequests.
    // This is to avoid a race condition where a device is polling while we
    // are posting the configs (already encrypted with the new keys)
```